apps/ledger-live-mobile/src/screens/CustomImage/Step3Transfer.tsx

```diff
@@ -51,129 +31,55 @@ This should NOT happen, it means that some data has been lost.`;
 const Step3Transfer: React.FC<
   StackScreenProps<ParamList, "CustomImageStep3Transfer">
 > = ({ route, navigation }) => {
-  const [reconstructedPreviewResult, setReconstructedPreviewResult] =
-    useState<ProcessorPreviewResult | null>(null);
-
-  const { rawData, previewData } = route.params;
-
-  const handlePreviewResult = useCallback((res: ProcessorPreviewResult) => {
-    setReconstructedPreviewResult(res);
-  }, []);
+  const { rawData, device: deviceFromRoute, previewData } = route.params;
+  const [device, setDevice] = useState<Device | null>(deviceFromRoute);
 
   const handleError = useCallback(
     (error: Error) => {
       console.error(error);
       navigation.navigate(
         ScreenName.CustomImageErrorScreen as "CustomImageErrorScreen",
-        { error },
+        { error, device },
       );
     },
-    [navigation],
+    [navigation, device],
   );
 
-  const previewDimensions = useMemo(
-    () =>
-      fitImageContain(
-        {
-          width: reconstructedPreviewResult?.width ?? 200,
-          height: reconstructedPreviewResult?.height ?? 200,
-        },
-        boxToFitDimensions,
-      ),
-    [reconstructedPreviewResult?.height, reconstructedPreviewResult?.width],
-  );
-
-  const insets = useSafeAreaInsets();
+  const completeAction = useCompleteActionCallback();
 
-  const [showReconstructed, setShowReconstructed] = useState(true);
-  const handlePressIn = useCallback(
-    () => setShowReconstructed(false),
-    [setShowReconstructed],
-  );
-  const handlePressOut = useCallback(
-    () => setShowReconstructed(true),
-    [setShowReconstructed],
-  );
-
-  const isDataMatching =
-    reconstructedPreviewResult?.imageBase64DataUri ===
-    previewData?.imageBase64DataUri;
+  const handleResult = useCallback(() => {
+    completeAction(PostOnboardingActionId.customImage);
+    navigation.getParent()?.goBack();
+  }, [navigation, completeAction]);
 
+  const insets = useSafeAreaInsets();
+  const DEBUG = false;
```

Comment:
Is this normal?

Reply:
hmmmmm I wanted to make it easy to debug in case their is something wrong with the image data